url7b90c1010b_DastanIqbal_HelloZMQwithJava_tgz-pJ8-ClientJ8/src/zmq/Trie.java

```diff
@@ -0,0 +1,298 @@
+/*
+    Copyright (c) 2009-2011 250bpm s.r.o.
+    Copyright (c) 2007-2009 iMatix Corporation
+    Copyright (c) 2011-2012 Spotify AB
+    Copyright (c) 2007-2011 Other contributors as noted in the AUTHORS file
+
+    This file is part of 0MQ.
+
+    0MQ is free software; you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    0MQ is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package zmq;
+
+
+public class Trie {
+    private int refcnt;
+
+    private byte min;
+    private short count;
+    private short live_nodes;
+    
+    public static interface ITrieHandler {
+        void added(byte[] data, int size, Object arg);
+    }
+    Trie[] next;
+    
+    public Trie() {
+        min = 0;
+        count = 0;
+        live_nodes = 0;
+        
+        refcnt = 0;
+        next = null;
+    }
+    
+    //  Add key to the trie. Returns true if this is a new item in the trie
+    //  rather than a duplicate.
+    public boolean add (byte[] prefix_)
+    {
+        return add (prefix_, 0);
+    }
+    
+    public boolean add (byte[] prefix_, int start_)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 96-112 into a separate method.